api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/security/SecurityConfiguration.java

```diff
@@ -96,10 +97,8 @@ protected void configure(HttpSecurity http) throws Exception {
 
             // login endpoint
             .and()
-            .addFilterBefore(
-                loginFilter(authConfigurationProperties.getServiceLoginEndpoint()),
-                UsernamePasswordAuthenticationFilter.class
-            )
+            .addFilterBefore(new ShouldBeAlreadyAuthenticatedFilter(authConfigurationProperties.getServiceLoginEndpoint(), handlerInitializer.getAuthenticationFailureHandler()), UsernamePasswordAuthenticationFilter.class)
```

Comment:
I am a bit confused here. Doesn't the ShouldBeAlreadyAuthenticatedFilter always stop processing when attempting authentication?

Reply:
Also another question why do you change it here in catalog?